sbc_compassion/models/sponsorship_correspondence.py

```diff
@@ -280,7 +280,13 @@ def _inverse_page(self):
 
     def _get_original_text(self, source_text):
         """ Gets the desired text (original/translated) from the pages. """
-        return '\n\n'.join(self.page_ids.mapped(source_text))
+        txt = self.page_ids.mapped(source_text)
```

Comment:
what about ```txt = self.page_ids.filtered(source_text).mapped(source_text)``` ?
https://www.odoo.com/documentation/8.0/reference/orm.html#openerp.models.Model.filtered
You won't need all further checks.